_studio/shared/umc/codec/h265_dec/src/umc_h265_slice_decoding.cpp

```diff
@@ -411,64 +608,58 @@ UMC::Status H265Slice::UpdateReferenceList(H265DBPList *pDecoderFrameList, H265D
+    for (i = 0; i < header->m_numPicStCurr0; i++)
     {
-        if(getRPS()->getUsed(i))
-        {
-            int32_t poc = header->slice_pic_order_cnt_lsb + getRPS()->getDeltaPOC(i);
+        int32_t poc = header->m_RpsPOCCurrList0[i];
 
-            H265DecoderFrame *pFrm = pDecoderFrameList->findShortRefPic(poc);
-            m_pCurrentFrame->AddReferenceFrame(pFrm);
+        H265DecoderFrame *pFrm = pDecoderFrameList->findShortRefPic(poc);
+        m_pCurrentFrame->AddReferenceFrame(pFrm);
 
-            if (pFrm)
-                pFrm->SetisLongTermRef(false);
-            RefPicSetStCurr0[NumPicStCurr0] = pFrm;
-            NumPicStCurr0++;
-            if (!pFrm)
-            {
-                /* Reporting about missed reference */
-                m_pCurrentFrame->SetErrorFlagged(UMC::ERROR_FRAME_REFERENCE_FRAME);
-                /* And because frame can not be decoded properly set flag "ERROR_FRAME_MAJOR" too*/
-                m_pCurrentFrame->SetErrorFlagged(UMC::ERROR_FRAME_MAJOR);
-            }
-            // pcRefPic->setCheckLTMSBPresent(false);
+        if (pFrm)
+            pFrm->SetisLongTermRef(false);
+        RefPicSetStCurr0[NumPicStCurr0] = pFrm;
+        NumPicStCurr0++;
+        if (!pFrm)
+        {
+            /* Reporting about missed reference */
+            m_pCurrentFrame->SetErrorFlagged(UMC::ERROR_FRAME_REFERENCE_FRAME);
+            /* And because frame can not be decoded properly set flag "ERROR_FRAME_MAJOR" too*/
+            m_pCurrentFrame->SetErrorFlagged(UMC::ERROR_FRAME_MAJOR);
         }
+        // pcRefPic->setCheckLTMSBPresent(false);
     }
 
-    for(; i < getRPS()->getNumberOfNegativePictures() + getRPS()->getNumberOfPositivePictures(); i++)
+    for (i = header->m_numPicStCurr0; i < header->m_numPicStCurr0 + header->m_numPicStCurr1; i++)
     {
-        if(getRPS()->getUsed(i))
-        {
-            int32_t poc = header->slice_pic_order_cnt_lsb + getRPS()->getDeltaPOC(i);
+        int32_t poc = header->m_RpsPOCCurrList0[i];
 
-            H265DecoderFrame *pFrm = pDecoderFrameList->findShortRefPic(poc);
-            m_pCurrentFrame->AddReferenceFrame(pFrm);
+        H265DecoderFrame *pFrm = pDecoderFrameList->findShortRefPic(poc);
+        m_pCurrentFrame->AddReferenceFrame(pFrm);
 
-            if (pFrm)
-                pFrm->SetisLongTermRef(false);
-            RefPicSetStCurr1[NumPicStCurr1] = pFrm;
-            NumPicStCurr1++;
-            if (!pFrm)
-            {
-                /* Reporting about missed reference */
-                m_pCurrentFrame->SetErrorFlagged(UMC::ERROR_FRAME_REFERENCE_FRAME);
-                /* And because frame can not be decoded properly set flag "ERROR_FRAME_MAJOR" too*/
-                m_pCurrentFrame->SetErrorFlagged(UMC::ERROR_FRAME_MAJOR);
-            }
-            // pcRefPic->setCheckLTMSBPresent(false);
+        if (pFrm)
+            pFrm->SetisLongTermRef(false);
+        RefPicSetStCurr1[NumPicStCurr1] = pFrm;
```

Comment:
just to double check: if pFram==null, you still need to assign it, i.e. nul to reference list and increase reference number?

Reply:
This is legacy logic, not changed by this fix, I can check further, but should not block this fix's code review and check-in.